core/modules/user/store/actions.ts

```diff
@@ -167,6 +167,7 @@ const actions: ActionTree<UserState, RootState> = {
    * Update user profile with data from My Account page
    */
   async update (_, profile: UserProfile) {
+    profile = userHooksExecutors.beforeUserProfileUpdate(profile)
```

Comment:
I wonder (I am not sure) - maybe should we send copy of `profile` to this function? Without that, `profile` would be mutable inside the `beforeUserProfileUpdate` - which could cause hard to debug side effects. Wdyt?

Reply:
I think it's more a question about how mutator should work. Right now mutator mutate object. Maybe mutator should then always return a copy? You can check that it used in that way in whole app :thinking: Everywhere its used always once, so the question is what will happen if we will use it more then once. Probably there will be random order of mutations

Reply:
Not good. We have to discuss it internally